### DISCUSSION TITLE
Offline management

### DISCUSSION THREAD

**Discussion by ScreamZ** [maplibre/maplibre-react-native#532] [Q&A]
Hello,

I've been successfully using offline manager to save tiles

```ts
 await MapLibreGL.offlineManager.createPack(
            {
              name: "offlinePack",
              styleURL:
                "https://api.maptiler.com/maps/voyager/style.json",
              minZoom: 14,
              maxZoom: 20,
              bounds: [
                [3.8833, 43.6119], // neLng, neLat for Montpellier
                [3.8133, 43.5919], // swLng, swLat for Montpellier
              ],
            },
            (offlineRegion, status) => console.log(offlineRegion, status),
            (offlineRegion, err) => console.log(offlineRegion, err)
          );
````

But any way to persist offline RasterSource for weather layer ?

I was wondering maybe i could download tileUrlTemplate of disk, but is there any way to use a local source instead of a remote source within `timeUrlTemplate`

Or can i build a dynamic styledURL including my Raster and will I have layer source downloaded too ?

Thanks

**Comment by KiwiKilian** [comment]
I've never tried for raster tiles, you could try adopting this, to use `file://` for your raster tiles https://github.com/maplibre/maplibre-react-native/discussions/591#discussioncomment-11736046.
